source/common/unified_malloc_framework/src/memory_provider.cpp

```diff
@@ -46,6 +54,42 @@ umfMemoryProviderCreate(struct umf_memory_provider_ops_t *ops, void *params,
     return UMF_RESULT_SUCCESS;
 }
 
+enum umf_result_t umfMemoryProviderRegister(umf_memory_provider_ops_t *ops) {
+
+    // TODO check if this provider isn't already registered
+    globalProviders.push_back(*ops);
+
+    return UMF_RESULT_SUCCESS;
+}
+
+enum umf_result_t
+umfMemoryProvidersRegistryGet(umf_memory_provider_ops_t *providers,
+                              size_t *numProviders) {
+
+    if (providers == NULL) {
+        *numProviders = globalProviders.size();
+    } else {
+        memcpy(providers, globalProviders.data(),
+               sizeof(umf_memory_provider_ops_t) * *numProviders);
+    }
+
+    return UMF_RESULT_SUCCESS;
+}
+
+// TODO rename ;)
+const umf_memory_provider_ops_t *umfMemoryProvidersRegistryGetOps(char *name) {
+    auto it = std::find_if(
+        std::begin(globalProviders), std::end(globalProviders),
+        [&](auto &ops) { return std::strcmp(ops.get_name(NULL), name) == 0; });
```

Comment:
A general comment: until now we assumed that all function pointers in `ops` would only be used through the public APIs of memory pool or provider, not accessed directly. This means that, you could assume the first argument (NULL in your case) is never NULL. If that's no longer the case we should probably add some info about this in documentation (for get_name at least)